velox/common/memory/ByteStream.cpp

```diff
@@ -18,6 +18,165 @@
 
 namespace facebook::velox {
 
+size_t ByteStream::size() const {
+  if (ranges_.empty()) {
+    return 0;
+  }
+  size_t total = 0;
+  for (auto i = 0; i < ranges_.size() - 1; ++i) {
+    total += ranges_[i].size;
+  }
+  return total + std::max(ranges_.back().position, lastRangeEnd_);
+}
+
+size_t ByteStream::remainingSize() const {
+  if (ranges_.empty()) {
+    return 0;
+  }
+  const auto rangeEnd = &ranges_[ranges_.size() - 1];
+  auto cur = current_;
+  size_t total = 0;
+  if (cur == rangeEnd) {
+    total += (std::max(ranges_.back().position, lastRangeEnd_) - cur->position);
+  } else {
+    total += cur->size - cur->position;
+  }
+  cur++;
```

Comment:
nit: ++cur; ditto

Reply:
What's the difference? We don't eval here.